src/components/Modals/Onboarding.vue

```diff
@@ -0,0 +1,742 @@
+<template>
+  <div class="onboarding">
+    <div class="onboarding-main">
+      <div class="onboarding-content">
+        <div class="onboarding-top">
+          <div v-if="!isEditingProfile" class="onboarding-header">👏 Welcome to CoursePlan</div>
+          <div v-if="isEditingProfile" class="onboarding-header">👋 Hi {{ user.firstName }}</div>
+          <div v-if="!isEditingProfile" class="onboarding-description">Let's get to know you first!</div>
+          <div v-if="isEditingProfile" class="onboarding-description">Let's edit your profile!</div>
+        </div>
+        <div class="onboarding-section">
+          <div class="onboarding-subHeader">Your Name</div>
+          <div class="onboarding-inputs onboarding-inputs--name">
+            <div class="onboarding-inputWrapper onboarding-inputWrapper--name">
+              <label class="onboarding-label">First Name*</label>
+              <input class="onboarding-input" v-model="firstName" />
+            </div>
+            <div class="onboarding-inputWrapper onboarding-inputWrapper--name">
+              <label class="onboarding-label">Middle Name</label>
+              <input class="onboarding-input" v-model="middleName" />
+            </div>
+            <div class="onboarding-inputWrapper onboarding-inputWrapper--name">
+              <label class="onboarding-label">Last Name*</label>
+              <input class="onboarding-input" v-model="lastName" />
+            </div>
+          </div>
```

Comment:
Doesn't firebase auth already give you these info?

Reply:
It's a discussion the design team has had before (and also again with me and Will recently). Letting them enter their own data enhances the 'personalized' feel of the platform. Evan also pointed out that this accounts for international students, those with nicknames, and members of the LGBTQ+.